src/onemod/stage/base.py

```diff
@@ -422,42 +422,42 @@ class ModelStage(Stage, ABC):
     """
 
     config: StageConfig
-    groupby: set[str] | None = None
-    _crossby: set[str] | None = None
-    _subset_ids: set[int] = set()
-    _param_ids: set[int] = set()
-    _collect_after: set[str] = set()
+    groupby: UniqueList[str] | None = None
+    _crossby: UniqueList[str] | None = None
+    _subset_ids: UniqueList[int | None] = list()
+    _param_ids: UniqueList[int | None] = list()
+    _collect_after: UniqueList[str] = list()
 
     @computed_property
-    def crossby(self) -> set[str] | None:
+    def crossby(self) -> UniqueList[str] | None:
         return self._crossby
 
     @property
-    def subset_ids(self) -> set[int]:
+    def subset_ids(self) -> UniqueList[int | None]:
         if self.groupby is not None and not self._subset_ids:
             try:
                 subsets = self.dataif.load("subsets.csv", key="output")
-                self._subset_ids = set(subsets["subset_id"])
+                self._subset_ids = list(set(subsets["subset_id"]))
```

Comment:
Hadn't thought about it before, but maybe we want `subset_ids` and `param_ids` to be sorted?